source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -62,15 +62,16 @@ function onAltClick(event: delegate.Event<MouseEvent, HTMLInputElement>): void {
 	});
 }
 
-function init(): VoidFunction {
-	// `mousedown` required to avoid mouse selection on shift-click
-	delegate(document, '.js-reviewed-toggle', 'mousedown', batchToggle);
-	delegate(document, '.js-toggle-user-reviewed-file-form', 'submit', remember);
-	delegate(document, '.js-reviewed-toggle', 'click', onAltClick);
-
-	return () => {
-		previousFile = undefined;
-	};
+function init(): Deinit[] {
+	return [
+		// `mousedown` required to avoid mouse selection on shift-click
+		delegate(document, '.js-reviewed-toggle', 'mousedown', batchToggle),
+		delegate(document, '.js-toggle-user-reviewed-file-form', 'submit', remember),
+		delegate(document, '.js-reviewed-toggle', 'click', onAltClick),
+		() => {
+			previousFile = undefined;
+		},
```

Comment:
Unrelated to this PR but it would be better if we don't need to deal with this global state. Maybe put it in some container's `[data-previous-file]` attribute and let PJAX destroy the container for us.

Reply:
I don't see a problem in this specific instance, the state is correctly erased on page change.

Reply:
It's not a _problem_, I just think [less global state](https://github.com/refined-github/refined-github/pull/5415/commits/5c773a941dbc16eb8f3185fafb3b16709706c013) is better generally. By requiring passing local state you avoid mistakenly relying on the global state when it doesn't actually reflect the current situation.

Reply:
If there's an alternative solution, sure, but if it doubles or triples the amount of code I'd rather not